docs/scenarios_authoring/DataSourcesAndSinks.md

```diff
@@ -34,8 +34,23 @@ The `kafka` sink configuration form will show a list of fields defined in Schema
 
 ![Kafka sink](img/kafkaSink.png "Kafka sink")
 
-If you prefer to define the whole output event as one expression containing value to be written to the Kafka topic, instead of filling separete autogenerated fields, you can do that by switching `Raw editor` to `true`.
+If you prefer to define the whole output event as one expression containing value to be written to the Kafka topic, instead of filling separate autogenerated fields, you can do that by switching `Raw editor` to `true`.
 
 When using the Raw editor, you can choose the validation mode:
-* strict - requires providing all fields, including optional fields, without redundant fields;
-* lax - requires providing only required fields, you can skip optional fields and add redundant fields.
\ No newline at end of file
+* strict - requires providing all fields, including optional fields, without redundant fields, requires exact matching;
+* lax - requires providing only required fields, you can skip optional fields and add redundant fields, exact matching is not required;
+
+(*) allowing/denying redundant fields works only in Avro schema. In JSON schema this configuration is provided
+explicitly by schema property: `additionalProperties`. See more about how to use [JSON Schema](/docs/scenarios_authoring/HandlingSchemas.md#json-schema).
+
+(*) exact matching describes how output validation works for types Unknown and Union. Sometimes helpers or enrichers can
+return data with type `Unknown` - it means we can't recognize the type, consequently, we have a situation difficult to handle.
+And it's the place where exact matching comes to help us:
+
+* strict mode doesn't allow to pass Unknown for any schema type. In a situation when we try to pass a value with type
+  union as the output with schema union, strict mode requires these two unions should be the same;
```

Comment:
I would begin with the explanation of the lax mode, because afterwords it may be easier to explain strict mode.

If it is the case that lax and strict modes differ mainly in their treatment of union, I wonder whether a link to jsons schema defitionion of union - or a nice article explaining it would help. It looks like a bit tricky thing, a quick search on Google did not return any link with contents formulated in a human understandable language. 

_strict mode requires these two unions should be the same_  --> I do not understand it at all. How a value can be of all the types used in the particular union definition. (Well, maybe I do not understand union).